src/apispec/core.py

```diff
@@ -287,14 +331,18 @@ def path(
         :param dict|None operations: describes the http methods and options for `path`
         :param str summary: short summary relevant to all operations in this path
         :param str description: long description relevant to all operations in this path
+        :param list|None parameters: list of parameters relevant in this path
         :param dict kwargs: parameters used by any path helpers see :meth:`register_path_helper`
         """
         operations = operations or OrderedDict()
+        parameters = parameters or []
 
         # Execute path helpers
         for plugin in self.plugins:
             try:
-                ret = plugin.path_helper(path=path, operations=operations, **kwargs)
+                ret = plugin.path_helper(
+                    path=path, operations=operations, parameters=parameters, **kwargs
+                )
```

Comment:
`BasePlugin.path_helper` signature and docstring should be updated in apispec/src/apispec/plugin.py

And the fact that `parameters` is passed should be tested in `tests.test_core.TestPlugins.test_plugin_path_helper_is_used`.

Reply:
On this point I think we should:

* Keeping current unit tests (to ensure that plugins doesn't break with this new signature)
* Test that we have access to `parameters` in kwargs with current signature
* Test new signature

What do you think ?

Reply:
I wouldn't bother. I'd just test the new signature.

The point is about breaking custom plugins that wouldn't accept `**kwargs`. Modifying the signature when inheriting is a serious smell, so one shouldn't be doing this and the change this PR introduces is arguably not a breaking one.

Yet, we can be extra-conservative and make it a major release. Before doing so, we should make sure to document the `**kwargs` in the docs about writing plugins and in `BasePlugin` docstring so that we're totally covered for next time.

Reply:
Sorry for not responding earlier I'm writing doc on that topic before updating this PR

Reply:
No problem. No rush. Been busy as well.

The doc issue is not directly related. Don't feel like you *have* to take care of it. But if you do, that's great. Thanks.

Reply:
Don't worry, I prefer this PR to be as complete as possible and it's part of the feature after all. I updated tests and documentation

Reply:
Thanks @karec for working on this. Just a few minor comments. We're almost done.